core/files/css/base.php

```diff
@@ -643,6 +643,28 @@ protected function is_update_required() {
 		return false;
 	}
 
+	/**
+	 * Clear stacks
+	 *
+	 * If a page contains templates, such as loop items, shortcodes, kits, or template widgets,
+	 * the templates responsive setting will not be updated unless the 'Regenerate CSS' option is used.
+	 * Reproduce only when additional breakpoints are active.
+	 *
+	 * @since 3.14.0
+	 *
+	 * @access private
+	 */
+	private function clear_stacks() {
```

Comment:
I think the full logic is not related to here, so please move the logic to the app/module layers.

You can work with hooks, for example, on `parse_content`, trigger an action: 
```
do_action( 'elementor/css_file/parse_content' );
```
Then from `additional_custom_breakpoints` experiment, run logic and then `clear_stack_cache`:
```
add_action( 'elementor/css_file/parse_content', function() { 
	if...
	Plugin::$instance->controls_manager->clear_stack_cache( $handle_id );
});
```